crates/nu-cli/src/commands/sort_by.rs

```diff
@@ -107,12 +133,38 @@ pub fn sort(
             value: UntaggedValue::Primitive(_),
             ..
         } => {
-            vec.sort_by(|a, b| coerce_compare(a, b).expect("Unimplemented BUG: What about primitives that don't have an order defined?").compare());
+            let should_sort_case_insensitively = insensitive && vec.iter().all(|x| x.is_string());
+
+            vec.sort_by(|a, b| {
+                if should_sort_case_insensitively {
+                    let lowercase_a_string = a.expect_string().to_ascii_lowercase();
+                    let lowercase_b_string = b.expect_string().to_ascii_lowercase();
```

Comment:
At first glance, it would seem that this would fail when running these `expect_string()`s while the input was not a string, but if you look a few lines up, we do a check to make sure all values are strings in the `.all()` iterator method, so this should never fail.